src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlBulkCopyTest/WriteToServerTest.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Data;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
+    public class WriteToServerTest
+    {
+        private string _connectionString = null;
+
+        public WriteToServerTest()
+        {
+            _connectionString = (new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) { MultipleActiveResultSets = true }).ConnectionString;
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsNotAzureServer))]
+        public async Task WriteToServerWithDbReaderFollowedByWriteToServerWithDataRowsShouldSucceed()
+        {
+            try
+            {
+                SetupTestTables();
+
+                DataRow[] dataRows = CreateDataRows();
+
+                bool result = DoBulkCopy(dataRows);
+                Assert.True(result, "WriteToServer with IDataReader followed by WriteToServer with data rows test failed.");
+                Assert.True(GetCopiedRecordsCount() == 8, "WriteToServer with IDataReader followed by WriteToServer with data rows test failed.");  
+
+                CleanupTestTable2();
+
+                result = await DoBulkCopyAsync(dataRows);
+                Assert.True(result, "WriteToServerAsync with IDataReader followed by WriteToServerAsync with data rows test failed.");
+                Assert.True(GetCopiedRecordsCount() == 8, "WriteToServerAsync with IDataReader followed by WriteToServerAsync with data rows test failed.");
+            }
+            finally
+            {
+                RemoveTestTables();
+            }
+        }
+
+        private void SetupTestTables()
+        {
+            using (SqlConnection connection = new SqlConnection(_connectionString))
+            {
+                connection.Open();
+
+                SqlCommand command = connection.CreateCommand();
+
+                Helpers.TryExecute(command, "drop table if exists TestTable1");
+                Helpers.TryExecute(command, "drop table if exists TestTable2");
```

Comment:
You'll want to use unique object names to avoid conflicts across runs. This is mainly an issue when tests are running against shared databases (e.g. Azure SQL, the AE environment).

Define some class level variables:
```
private string _tableName1 = DataTestUtility.GetUniqueName("Bulk1");
private string _tableName2 = DataTestUtility.GetUniqueName("Bulk2");
```

Reply:
The suggestion was applied to the unit test.